beacon_node/store/src/config.rs

```diff
@@ -210,28 +225,70 @@ impl StoreItem for OnDiskStoreConfig {
     }
 
     fn as_store_bytes(&self) -> Vec<u8> {
-        self.as_ssz_bytes()
+        match self {
+            // V0 is not prefixed explicitly, but unless the user chose a crazy configuration the
+            // serialized value will have a leading 0 byte.
+            OnDiskStoreConfig::V0(value) => value.as_ssz_bytes(),
+            OnDiskStoreConfig::V1(value) => {
+                let mut out = value.as_ssz_bytes();
+                out.insert(0, 1);
```

Comment:
Rather than doing this, why don't we do it in a slightly more compositional way by adding a `u8` field to the `OnDiskStoreConfig` for `V22` and later? Like:

```rust
struct OnDiskStoreConfig {
    #[only(V1)]
    slots_per_restore_point: u64,
    #[only(V22)]
    version_byte: u8,
    #[only(V22)]
    hierarchy_config: HierarchyConfig,
}
```

Then we can slightly reduce the amount of manual byte manipulation we need to do.

Reply:
Fixed in e5661720685236e73c6fee55998f15ba44a9ed88